File: 800/Java/P18.java
import java.util.*;

public class P18 {
    static void solve(Scanner scanner) {
        int x = scanner.nextInt();
        int k = scanner.nextInt();

        // if we can directly go to the destination
        if(x % k != 0) {
            System.out.println(1);
            System.out.println(x);
        } else {
            // we can reach destination in 2 steps
            System.out.println(2);
            System.out.print(x-1 + " ");
            System.out.println(1);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while(t-- > 0)  solve(scanner);
        scanner.close();
    }
}

File: 800/C++/p18.cpp
#include<bits/stdc++.h>
using namespace std;

void solve() {
    int x, k;
    cin>>x>>k; 
    // if we can directly go to the destination
    if(x % k != 0) {
        cout<<1<<endl;
        cout<<x<<endl;
    } else {
        // we can reach destination in 2 steps
        cout<<2<<endl;
        cout<<x-1<<" ";
        cout<<1<<endl;
    }
}

int main() {
    int t;
    cin>>t;
    while(t--)  solve();
    return 0;
}